doc/api/process.markdown

```diff
@@ -16,6 +16,34 @@ continue.
 as [`process.exit()`][] or uncaught exceptions, and should not be used as an
 alternative to the `'exit'` event unless the intention is to schedule more work.
 
+## Event: 'deprecation'
```

Comment:
@Fishrock123 it wasn't just because of userland like `depd`, but also because there are already `--*-deprecation` command line switches, and this is adding `--*-warnings` command line switches. Having two events also aligns what those switches are controlling better, i.m.o.

For example, if you run `node --no-deprecation`, you can better understand which events won't be emitted any longer, for example, rather than it reducing the events emitted on a differently-named event and `--no-warnings` not actually turning off all "warnings".